src/accountsdb/db.zig

```diff
@@ -3164,11 +3246,20 @@ pub fn indexAndValidateAccountFile(
     }
 
     while (true) {
-        const account = accounts_file.readAccount(offset) catch break;
+        const account = accounts_file.readAccount(
+            allocator,
+            buffer_pool,
+            offset,
+        ) catch |err| switch (err) {
+            error.EOF => break,
+            else => |e| return e,
+        };
+        defer account.deinit(allocator);
+
         try account.validate();
 
         if (geyser_storage) |storage| {
-            try storage.cloneAndTrack(account);
+            try storage.cloneAndTrack(allocator, account);
```

Comment:
just a note we probably gotta come up with a better allocation strategy here -- ideally prealloc a batch of mem for account data because the alloc calls will slow it down a lot

Reply:
045fd3a274689c7eb2eca57426831317847a20ea

Reply:
not sure what this commit solves?

Reply:
In this codepath, the data is now not cloned, the allocator is just for copying the frames slice. Though this doesn't reduce the number of allocations, which is unfortunate. Maybe a better way of sharing ReadHandles is in order